### ISSUE TITLE
Type inference error after upgrading to version 1.4.x

### ISSUE THREAD

**Issue by Mini-ghost** [vuejs/language-tools#2679]
## Description

After upgrading vue-tsc to version 1.4.1, type checking failed and this issue still exists even in version 1.4.4.

I defined a `ComponentProps<C>` type in global.d.ts as follows:

```ts
import type { VNodeProps } from 'vue';

declare global {
  type ComponentProps<C> = Omit<InstanceType<C>['$props'], keyof VNodeProps>;
}
```

This is to make it easier for me to obtain the type of props that needs to be used to other components. I use it like this:

```vue
<script setup lang="ts">
import { computed } from 'vue';
import AppKanban from '@/components/molecules/AppKanban.vue';

const columns = computed<ComponentProps<typeof AppKanban>['columns']>(() => [
  // columns setting
])
</script>

<template>
  <AppKanban :columns="columns" />
</template>
```

Currently, the latest version of Volar does not show any errors and correctly infers types, but when running type checking, it fails with the following error message:

```
src/pages/grade/system/index.vue:63:7  - error TS2322: Type 'unknown' is not assignable to type 'AppKanbanColumnItem[]'.
```

<img width="888" alt="截圖 2023-04-23 上午10 07 58" src="https://user-images.githubusercontent.com/39984251/233815645-8f9bd64d-d6ca-403e-9c37-5d53cf00d89d.png">

This issue did not exist in version 1.2.x.

**Comment by johnsoncodehk** [comment]
Is the TS version in your project consistent with VSCode? If yes, please provide a repro case so we can investigate it.

**Comment by Mini-ghost** [comment]
Thank you for your response. The following link contains a reproduction of the issue:

https://stackblitz.com/edit/vitejs-vite-f2muxx?file=src%2FApp.vue,src%2Fcomponents%2FAppKanban.vue

I have attempted various methods to replicate the issue and have found that it only occurs when using `<slot>`. Once I removed the `<slot>` from the reproduction, the issue was resolved.

**Comment by johnsoncodehk** [comment]
Fixed in v1.5.3.
